opentelemetry-sdk/src/metrics/internal/sum.rs

```diff
@@ -7,12 +7,36 @@ use std::{sync::Mutex, time::SystemTime};
 use crate::metrics::data::{self, Aggregation, DataPoint, Temporality};
 use opentelemetry::KeyValue;
 
-use super::{AtomicTracker, Number};
-use super::{Increment, ValueMap};
+use super::{Aggregator, AtomicTracker, Number};
+use super::{AtomicallyUpdate, ValueMap};
+
+struct Increment<T>
+where
+    T: AtomicallyUpdate<T>,
+{
+    value: T::AtomicTracker,
+}
+
+impl<T> Aggregator<T> for Increment<T>
+where
+    T: Number,
+{
+    type Config = T;
```

Comment:
Could we have `Config` defined as a zero-sized type?

```rust
impl<T> Aggregator<T> for Increment<T>
where
    T: Number,
{
    type Config = ();

    fn create(_init: &()) -> Self {
        Self {
            value: T::new_atomic_tracker(T::default()),
        }
    }

    fn update(&self, _init: &(), measurement: T) {
        self.value.add(measurement)
    }
}
```
It makes more sense semantically as `Increment` and `Assign` based updates do not truly require `Config`. Also, I'm assuming it should have lesser memory footprint (lesser stack frame size) than using a `u64` or `f64`.